libraries/chain/database.cpp

```diff
@@ -153,9 +155,17 @@ void database::open( const open_args& args )
 
 uint32_t database::reindex( const open_args& args )
 {
+   bool reindex_success = false;
+   uint32_t last_block_number = 0; // result
+
+   BOOST_SCOPE_EXIT(this_,&reindex_success,&last_block_number) {
```

Comment:
Is there a use case, today, for a plugin doing something different on reindex failure? (Specifically the AH RocksDB plugin)

If not, I think this is being clever for the sake of being clever and the reindex success signal should be used when the reindex succeeds.

Reply:
Reindex-done signal shall be always sent to avoid reconfiguration of a plugin after finished reindex (RocksDB-AH plugin needs it to change data flushing strategy).
It can be also helpful for generating a report of time spent in plugin during reindex process.